packages/filesystem/src/browser/file-upload-service.ts

```diff
@@ -76,14 +77,15 @@ export class FileUploadService {
 
         fileInput.addEventListener('change', () => {
             if (this.deferredUpload && fileInput.value) {
-                const body = new FormData(form);
+                const source = new FormData(form);
                 // clean up to allow upload to the same folder twice
                 fileInput.value = '';
-                const targetUri = new URI(<string>body.get(FileUploadService.TARGET));
+                const targetUri = new URI(<string>source.get(FileUploadService.TARGET));
                 const { resolve, reject } = this.deferredUpload;
                 this.deferredUpload = undefined;
+                const { onDidUpload } = this.uploadForm;
```

Comment:
Q: where is `onDidUpload` declared?

Reply:
https://github.com/theia-ide/theia/pull/5147/files/f245e700acf2eca57ffdb06185f76430d2ba658a#diff-02495d5f1e3aa572d518608c8a1a88d4R107

Reply:
found it, nvm!